### PR TITLE
Add isValid(), fix security issues in dependency

### PR DESCRIPTION
**Pull request by bluet**

### CODE REVIEW

File: source/ulid.ts [hunk 1]
```diff
@@ -241,3 +241,15 @@ export function ulid(seedTime?: number, prng?: PRNG): ULID {
...
+
+// isValid() is a function that checks if a given string is a valid ULID.
+// checks:
+// - is a string
+// - every character is in the ENCODING string, and is the correct length
+export function isValid(id: string): boolean {
+    return (
+        typeof id === "string" &&
+        id.length === TIME_LEN + RANDOM_LEN &&
+        id.split("").every(char => ENCODING.indexOf(char) !== -1)
```
Comment (perry-mitchell) [inline source/ulid.ts:253]: ULID's are case insensitive, and this doesn't take into account that the passed `id` is in lowercase format. Perhaps a simple `toUpperCase` would suffice here?

File: test/ulid.spec.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const { expect } = require("chai");
+const { describe } = require("node:test");
```
Comment (perry-mitchell) [inline test/ulid.spec.js:2]: This can be removed.

File: test/ulid.spec.js [hunk 3]
```diff
@@ -190,4 +192,22 @@ describe("ulid", function() {
...
+            expect(isValid("01ARYZ6S41TSV4RRFFQ69G5FA")).to.be.false;
+        });
+
+        it("should return false for invalid ULIDs (wrong characters)", function() {
+            expect(isValid("01ARYZ6S41TSV4RRFFQ69G5FA!")).to.be.false;
+        });
+
+        it("should return false for invalid ULIDs (wrong type)", function() {
+            expect(isValid(123)).to.be.false;
+        });
```
Comment (perry-mitchell) [inline test/ulid.spec.js:211]: Please add a test here for a lowercase ULID.

**Review by perry-mitchell — CHANGES_REQUESTED** [review]
Couple of small things. Thanks for this!

File: source/ulid.ts (hunk 1 above)
Comment (bluet) [reply source/ulid.ts]: added .toUpperCase()

File: test/ulid.spec.js (hunk 2 above)
Comment (bluet) [reply test/ulid.spec.js]: Removed. sorry I didn't notice when and why it's been added lol
Maybe it's from the vscode extension which adds missing import automatically for me.

File: test/ulid.spec.js (hunk 3 above)
Comment (bluet) [reply test/ulid.spec.js:214]: lowercase test added

**Comment by bluet** [comment]
hi @perry-mitchell patches pushed :+1: 

hope I can use the new version of ulidx from npm instead of my local build hahaha

**Comment by perry-mitchell** [comment]
Thanks, I'll check this in a few hours.. but I suspect it'll be fine and I'll do a release then.

Cheers!

**Review by perry-mitchell — APPROVED** [review]

**Pull request merged**